### PR TITLE
Chinese version for C Coding Style Guide

### PR DESCRIPTION
**Pull request by jiguorui**
I have translate the OpenResty® C Coding Style Guide into Chinese.

### CODE REVIEW

**Comment by agentzh** [comment]
Wow, this is awesome. We'll review this thing as soon as we can manage. Many thanks!

File: v2/cn/c-coding-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,880 @@
...
+`p`, `q`, 等等. 这些变量通常生命周期很短并且作用域有限. 根据霍夫曼原则, 我们应该在一些常见用法里使用简称命名
+以避免文字杂音.即使是简称命名也要遵循 NGINX 的习惯. 如无必要别发明自己的命名方式. 一定要用有意义的名字. 即使
+像 `p` 和 `q`,它们是在上下文中进行字符处理的字符串指针变量的常见命名.
+
+C 结构体和联合体的名字应该尽可能地用单词的全拼形式(除非名称过于长). 举例子, 在 NGINX core 里 `struct ngx_http_request_s`, 
+我们有长的成员名字 `read_event_handler`, `upstream_states`, 和 `request_body_in_persistent_file`.
+
+# 缩进
+
+在 NGINX 的世界里用并且只用 *空格* 作为缩进. 不要用 tabs! 一般我们用 *4-space* 缩进 除了有一些特殊的
```
Comment (agentzh) [inline v2/cn/c-coding-style-guide.md]: `tabs` => `制表符`
`4-space` => `四空格`

Please fix other similar places. Thank you.

**Comment by agentzh** [comment]
@jiguorui Sorry, I just committed some changes to the English version today in commit 9fa7554fee. Will you please sync those changes to your Chinese version here as well?

Many thanks!

File: v2/cn/c-coding-style-guide.md (hunk 1 above)
Comment (jiguorui) [reply v2/cn/c-coding-style-guide.md]: Sure, I will

File: v2/cn/c-coding-style-guide.md [hunk 2]
```diff
@@ -0,0 +1,911 @@
...
+
+OpenResty 在它的 C 语言模块里遵循 NGINX 的代码风格, 比如 OpenResty 自己的那些 NGINX 插件
+模块和 OpenResty 的那些 Lua 库的 C 部分. 不太幸运的是，即使是 NGINX core 自己的 C 代码
+也没有严格遵循和其他 code base 同样的代码习惯. 能够有一个正式的导则文档以避免混淆是一个非常
+被期待的事情.
+
+给 OpenResty core 提交的补丁也应该遵循这些导则，不然它们将会无法通过代码审核也无法得到合并.
+OpenResty 和 NGINX 社区也鼓励大家在用 C 开发插件和库的时候去遵循这个导则.
+
+# 命名习惯
```
Comment (agentzh) [inline v2/cn/c-coding-style-guide.md]: "习惯" => "约定".

File: v2/cn/c-coding-style-guide.md [hunk 3]
```diff
@@ -0,0 +1,911 @@
...
+#define ngx_conf_init_value(conf, default)
+\
+    if (conf == NGX_CONF_UNSET) {                                            \
+        conf = default;                                                      \
+    }
+```
+
+我们推荐放置 \ 在第 78 列, 尽管有时侯 NGINX core 也没有完全这样做.
+
+# Global/Static 变量
```
Comment (agentzh) [inline v2/cn/c-coding-style-guide.md]: Should be "全局/静态变量".

File: v2/cn/c-coding-style-guide.md [hunk 4]
```diff
@@ -0,0 +1,911 @@
...
+还有正确缩进相关的配置:
+
+```vim
+set expandtab
+set shiftwidth=4
+set softtabstop=4
+set tabstop=4
+```
+
+# Goto 语句和代码标记
```
Comment (agentzh) [inline v2/cn/c-coding-style-guide.md]: "标记" => "标签"

**Comment by agentzh** [comment]
Merged with translations for the title part too. Many thanks!

**Comment by agentzh** [comment]
@jiguorui BTW, I noted that the English full-stop punctuation is used instead of the Chinese one (`。`). It'll be great if can create a separate pull request to address this in this CN article. Many thanks!

**Comment by jiguorui** [comment]
@agentzh I will address the issue about full-stop punctuations in these days.

**Pull request closed without merging**